File: Problem-021/solution.js
function sumOfDivisors(n) {
    let sum = 0;

    for (let i = 1; i < n; i++) {
        if (n % i === 0) {
            sum += i;
        }
    }

    return sum;
}

function sumAmicableNum(limit) {
    let total = 0;

    for (let a = 1; a < limit; a++) {

        let b = sumOfDivisors(a);

        if (b !== a && b < limit) {

            if (sumOfDivisors(b) === a) {
                total += a;
            }
        }
    }

    return total;
}

File: Problem-021/solution.py
def sum_of_divisors(n):
    total = 0

    for i in range(1, n):
        if n % i == 0:
            total += i

    return total


def amicable_numbers(limit):
    total = 0

    for a in range(1, limit):

        b = sum_of_divisors(a)

        if b != a and b < limit:
            if sum_of_divisors(b) == a:
                total += a

    return total


